fanctrl.py

```diff
@@ -11,12 +12,245 @@
+            "--list-strategies",
+            action="store_true"
+        )
+        commandGroup.add_argument(
+            "--reload",
+            "-r",
+            action="store_true"
+        )
+        commandGroup.add_argument(
+            "--pause",
+            action="store_true"
+        )
+        commandGroup.add_argument(
+            "--resume",
+            action="store_true"
+        )
+        commandGroup.add_argument(
+            "--hardware-controller",
+            "--hc",
+            type=str,
+            choices=["ectool"],
+            default="ectool"
+        )
+        commandGroup.add_argument(
+            "--socket-controller",
+            "--sc",
+            type=str,
+            choices=["unix"],
+            default="unix"
+        )
+
+    def parseArgs(self, args=None):
+        values = None
+        original_stderr = sys.stderr
+        # silencing legacy parser output
+        sys.stderr = open(os.devnull, 'w')
+        try:
+            legacy_values = self.legacyParser.parse_args(args)
+            if legacy_values.strategy is None:
+                legacy_values.strategy = legacy_values._strategy
+            # converting legacy values into new ones
+            values = argparse.Namespace()
+            values.socket_controller = legacy_values.socket_controller
+            if legacy_values.query:
+                values.command = "print"
+                values.print_selection = "current"
+            if legacy_values.list_strategies:
+                values.command = "print"
+                values.print_selection = "list"
+            if legacy_values.resume:
+                values.command = "resume"
+            if legacy_values.pause:
+                values.command = "pause"
+            if legacy_values.reload:
+                values.command = "reload"
+            if legacy_values.run:
+                values.command = "run"
+                values.silent = legacy_values.no_log
+                values.hardware_controller = legacy_values.hardware_controller
+                values.config = legacy_values.config
+                values.strategy = legacy_values.strategy
+            if not hasattr(values, "command") and legacy_values.strategy is not None:
+                values.command = "use"
+                values.strategy = legacy_values.strategy
+            if not hasattr(values, "command"):
+                raise Exception("not a valid legacy command")
+            if self.isRemote or values.command == "run":
+                print(
+                    "[Warning] > this command is deprecated and will be removed soon, please use the new command format instead ('fw-fanctrl -h' for more details).")
+        except (SystemExit, Exception):
+            sys.stderr = original_stderr
+            values = self.parser.parse_args(args)
+        finally:
+            sys.stderr = original_stderr
+        return values
+
+
+class JSONException(Exception):
```

Comment:
you re-declared JSONException and UnimplementedException

Reply:
Done